### PR TITLE
Paginate the things

### PR DESCRIPTION
**Pull request by rjstires**
## Purpose
Volumes, domains, and Node Balancers lacked pagination. Now they don't!

## Gold Platting
- Updated all three landings to use a similar render pattern.
- The features "appear" to render much more quickly, rather than just staring at a spinner for x seconds.
- I would like to follow this up with CSS skeleton loading designs. (https://codepen.io/oslego/pen/XdvWmd)

### CODE REVIEW

File: src/features/NodeBalancers/NodeBalancersLanding.tsx [hunk 1]
```diff
@@ -157,16 +185,13 @@ export class NodeBalancersLanding extends React.Component<CombinedProps, State>
...
+        this.setState({
+          nodeBalancers: this.state.nodeBalancers.filter((nodebalancer) => nodebalancer.id === selectedNodeBalancerId),
+          deleteConfirmDialog: {
+            open: false,
+            submitting: false,
+          },
+        });
       })
       .catch((err) => {
         const apiError = path<Linode.ApiFieldError[]>(['response', 'data', 'error'], err);
```
Comment (Jskobos) [inline src/features/NodeBalancers/NodeBalancersLanding.tsx]: Think submitting should be set to false even on error so the user can re-submit

File: src/features/NodeBalancers/NodeBalancersLanding.tsx [hunk 2]
```diff
@@ -157,16 +185,13 @@ export class NodeBalancersLanding extends React.Component<CombinedProps, State>
 
     deleteNodeBalancer(selectedNodeBalancerId!)
       .then((response) => {
-        getNodeBalancersWithConfigs()
-          .then((response: Linode.ExtendedNodeBalancer[]) => {
-            this.setState({
-              nodeBalancers: response,
```
Comment (Jskobos) [inline src/features/NodeBalancers/NodeBalancersLanding.tsx]: I think this was necessary, deleted NBs now hang around on the list until reload

Comment (rjstires) [reply src/features/NodeBalancers/NodeBalancersLanding.tsx]: Quite the opposite, the filter was reversed so only the deleted NB was staying around. Good eye.

File: src/features/NodeBalancers/NodeBalancersLanding.tsx (hunk 1 above)
Comment (rjstires) [reply src/features/NodeBalancers/NodeBalancersLanding.tsx]: Good catch.

**Comment by alioso** [comment]
Styling done and added a new, reusable skeleton loader that I applied for tables. We can start adding types of loaders as we see fit

**Comment by alioso** [comment]
@rjstires I am keeping this as "change requested" as I believe the pager and show dropdown should not be shown if under 25 or whatever requirement is in place. It is confusing from a UI perspective to show a pager if it has no actionable value,

**Pull request merged**